drivers/include/lps331ap.h

```diff
@@ -62,18 +54,23 @@ typedef struct {
     lps331ap_rate_t rate;       /**< tell sensor to sample with this rate */
 } lps331ap_params_t;
 
+/**
+ * @brief   Device descriptor for LPS331AP sensors
+ */
+typedef struct {
+    lps331ap_params_t params;   /**< device initialization parameters */
+} lps331ap_t;
+
 /**
  * @brief   Initialize a given LPS331AP pressure sensor
  *
  * @param[out] dev      device descriptor of the sensor
- * @param[in]  i2c      I2C bus the sensor is connected to
- * @param[in]  address  the sensor's address on the I2C bus
- * @param[in]  rate     internal sampling rate of the sensor
+ * @param[in]  params   initialization parameters
  *
  * @return              0 on success
  * @return              -1 on error
  */
-int lps331ap_init(lps331ap_t *dev, i2c_t i2c, uint8_t address, lps331ap_rate_t rate);
+int lps331ap_init(lps331ap_t *dev, const lps331ap_params_t * params);
```

Comment:
Here you have an API change, I'd say you should make another PR for these.

Reply:
This is about unifying the param scheme in _all_ drivers, so I'd say it fits in the goal of this PR

Reply:
Yeah but here you're not modifying the params scheme but the API for this driver. I see it takes the params as input but what I argue here is that you modified the API, the params and structs just to fit with the new params scheme. Too much changes in a row...

Reply:
> you modified the API, the params and structs just to fit with the new params scheme

Which again is the goal of this PR: unifying the way initialization parameters are defined **and** used.

> Too much changes in a row...

Maybe but required